cpp/src/arrow/array/array_test.cc

```diff
@@ -3348,4 +3358,70 @@ TEST(TestSwapEndianArrayData, InvalidLength) {
   }
 }
 
+template <typename T>
+class TestPrimitiveArray : public ::testing::Test {
+ public:
+  void SetUp() { pool_ = default_memory_pool(); }
+
+ protected:
+  MemoryPool* pool_;
+};
+
+typedef ::testing::Types<PBoolean, PUInt8, PUInt16, PUInt32, PUInt64, PInt8, PInt16,
+                         PInt32, PInt64, PFloat, PDouble>
+    NumericPrimitives;
```

Comment:
Is it possible to also make it work for `MonthIntervalType` / `MonthIntervalArray`? See `arrow/type_fwd.h` for their definitions.

Reply:
Well, I was referring to that itself in the first sentence of my [earlier comment](https://github.com/apache/arrow/pull/11694#discussion_r756153599) when I said two kinds of vectors are needed: 1) Array of scalars for all numeric types 2) Array of tuples is needed for IntervalArray types.

I tried and failed with one approach earlier. But I think current one(about to push) works fine. Please have a look.